tensorflow/compiler/xla/service/gpu/cudnn_conv_rewriter.cc

```diff
@@ -526,8 +526,8 @@ StatusOr<bool> RunOnInstruction(HloInstruction* conv) {
               conv->shape().element_type() != F32) {
         LOG(FATAL)
             << "The convolution instruction with integer inputs only allows "
-               "float outputs.  Insert a convert "
-               "instruction after the convolution instruction for interger outputs.";
+               "float outputs.  Insert a clamp instruction with range [-128, 127) followed by a convert "
```

Comment:
At first, we thought a float-to-int8 conversion would be sufficient.  However, because XLA's float-to-int8 conversion doesn't saturate for overflow, we now look for a custom-call-convolution -> clamp(-128, 127) -> Convert-to-int8 pattern.

Reply:
Do you mean that the front-end (e.g. TensorFlow) will emit a custom-call instruction?

Reply:
No, custom-calls are from cudnn_convolution_rewriter or the first traversal of cudnn_fused_convolution_rewriter.  I mentioned it here, because it is part of the pattern.  After this transform, the clamp/convert are removed and the custom-call is modified to have int8 output.  Later, cudnn convolution is called with int8 output descriptor.